From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Thu, 27 Oct 2016 21:51:10 +0200
Subject: [PATCH] AArch64 support: api, cli, controllers, drivers

Enable new architecture 'aarch64' ('arm64' on Ubuntu).
Tested with DEB driver.

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 packetary/api/repositories.py       | 2 +-
 packetary/cli/commands/base.py      | 2 +-
 packetary/controllers/repository.py | 2 +-
 packetary/drivers/deb_driver.py     | 2 ++
 4 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/packetary/api/repositories.py b/packetary/api/repositories.py
index ec8c54c..2e826d7 100644
--- a/packetary/api/repositories.py
+++ b/packetary/api/repositories.py
@@ -73,7 +73,7 @@ class RepositoryApi(object):
 
         :param config: the configuration
         :param repotype: the kind of repository(deb, yum, etc)
-        :param repoarch: the architecture of repository (x86_64 or i386)
+        :param repoarch: the architecture of repository (x86_64, i386 or aarch64)
         """
         context = config if isinstance(config, Context) else Context(config)
         return cls(RepositoryController.load(context, repotype, repoarch))
diff --git a/packetary/cli/commands/base.py b/packetary/cli/commands/base.py
index 83c2f38..c2cc497 100644
--- a/packetary/cli/commands/base.py
+++ b/packetary/cli/commands/base.py
@@ -56,7 +56,7 @@ class BaseRepoCommand(BaseCommand):
             '-a',
             '--arch',
             type=str,
-            choices=["x86_64", "i386"],
+            choices=["x86_64", "i386", "aarch64"],
             metavar='ARCHITECTURE',
             default="x86_64",
             help='The target architecture.')
diff --git a/packetary/controllers/repository.py b/packetary/controllers/repository.py
index 921a5c1..4ed6883 100644
--- a/packetary/controllers/repository.py
+++ b/packetary/controllers/repository.py
@@ -45,7 +45,7 @@ class RepositoryController(object):
 
         :param context: the context
         :param driver_name: the name of required driver
-        :param repoarch: the architecture of repository (x86_64 or i386)
+        :param repoarch: the architecture of repository (x86_64, i386 or aarch64)
         """
         if cls._drivers is None:
             cls._drivers = stevedore.ExtensionManager(
diff --git a/packetary/drivers/deb_driver.py b/packetary/drivers/deb_driver.py
index ece9fa1..0df1aa4 100644
--- a/packetary/drivers/deb_driver.py
+++ b/packetary/drivers/deb_driver.py
@@ -48,10 +48,12 @@ _OPERATORS_MAPPING = {
 }
 
 _ARCHITECTURES = {
+    "aarch64": "arm64",
     "x86_64": "amd64",
     "i386": "i386",
     "source": "Source",
     "amd64": "x86_64",
+    "arm64": "aarch64",
 }
 
 _PRIORITIES = {
